core/src/main/scala/cr/pulsar/Consumer.scala

```diff
@@ -108,20 +108,17 @@ object Consumer {
       E: Inject[*, Array[Byte]]
   ](
       c: Consumer[F],
-      logAction: Array[Byte] => Topic.URL => F[Unit]
+      logAction: E => Topic.URL => F[Unit]
   ): Pipe[F, Message[Array[Byte]], E] =
     _.evalMap { m =>
-      val id   = m.getMessageId
-      val data = m.getData
+      val id = m.getMessageId
 
-      val acking = E.prj(data) match {
+      E.prj(m.getData) match {
         case Some(e) =>
-          c.ack(id).as(e)
+          logAction(e)(Topic.URL(m.getTopicName)) &> c.ack(id).as(e)
```

Comment:
It seems now we don't log incoming messages in case of decoding error, which could be useful. I think we should log every message that comes in

Reply:
We can add the raw data in the exception we are raising in case of decoding failure if that sounds okay? but we don't have access to a logger here and I don't think we should have one

Reply:
I mean we should also call `logAction` for `case None => ` so that we see what went in

Reply:
The types don't really match in that case. How would you do it?

Reply:
What I think we should do is to add the raw data to the exception message:

```scala
(new IllegalArgumentException(s"Decoding error: ${m.getData}")).raiseError[F, E]
```

And the caller should be responsible for handling this error and log it if necessary

Reply:
Missed that now you are logging the decoded value. We could also do `logAction(m.getData)` to see what came in, as we might want to see raw input rather than decoded value. What do you think? 
Note, if we would use binary messages instead of JSON it wouldn't make a lot of sense to log raw input